homeassistant/helpers/script.py

```diff
@@ -427,11 +427,12 @@ async def _async_delay_step(self):
 
         delay = delay.total_seconds()
         self._changed()
+        trace_set_result(delay=delay, done=False)
         try:
             async with async_timeout.timeout(delay):
                 await self._stop.wait()
         except asyncio.TimeoutError:
-            pass
+            trace_set_result(delay=delay, done=True)
```

Comment:
Why set delay again?
```suggestion
            trace_set_result(done=True)
```

Reply:
`trace_set_result` doesn't update a previously set result, it overwrites it.
We could rename the method to `trace_update_result` and change the implementation to update instead?